src/scenarios/shared/memoryconsumption.py

```diff
@@ -0,0 +1,165 @@
+from shared.util import helixworkitempayload, helixuploaddir, builtexe, publishedexe, uploadtokenpresent, getruntimeidentifier, iswin
+from shared.const import *
+from shared.testtraits import TestTraits
+from subprocess import CalledProcessError
+class MemoryConsumptionWrapper(object):
+    '''
+    Wraps memoryconsumption.exe, building it if necessary.
+    '''
+    def __init__(self):
+        memoryconsumptiondir = 'memoryconsumption'
+        self.reportjson = os.path.join(TRACEDIR, 'perf-lab-report.json')
+        if helixpayload() and os.path.exists(os.path.join(helixpayload(), memoryconsumptiondir)):
+            self._setmemoryconsumptionpath(os.path.join(helixpayload(), memoryconsumptiondir))
+        elif helixworkitempayload() and os.path.exists(os.path.join(helixworkitempayload(), memoryconsumptiondir)):
+            self._setmemoryconsumptionpath(os.path.join(helixworkitempayload(), memoryconsumptiondir))
+        else:
+            relpath = os.path.join(get_artifacts_directory(), memoryconsumptiondir)
+            memoryconsumptionproj = os.path.join('..',
+                                       '..',
+                                       'tools',
+                                       'ScenarioMeasurement',
+                                       'MemoryConsumption',
+                                       'MemoryConsumption.csproj')
+            memoryconsumption = CSharpProject(CSharpProjFile(memoryconsumptionproj,
+                                                   sys.path[0]),
+                                                   os.path.join(os.path.dirname(memoryconsumptionproj),
+                                                   os.path.join(get_artifacts_directory(), memoryconsumptiondir)))
+            if not os.path.exists(relpath):
+                memoryconsumption.restore(get_packages_directory(),
+                                True,
+                                getruntimeidentifier())
+                memoryconsumption.publish('Release',
+                                relpath,
+                                True,
+                                get_packages_directory(),
+                                None,
+                                getruntimeidentifier(),
+                                None,
+                                '--no-restore'
+                                )
+            self._setmemoryconsumptionpath(memoryconsumption.bin_path)
+
+    
+    def _setmemoryconsumptionpath(self, path: str):
+        self.memoryconsumptionpath = os.path.join(path, "memoryconsumption%s" % extension()) 
+
+    def parsetraces(self, traits: TestTraits):
+        directory = TRACEDIR
+        if traits.tracefolder:
+            directory = TRACEDIR + '/' + traits.tracefolder
+            getLogger().info("Parse Directory: " + directory)
+
+        memoryconsumption_args = [
+            self.memoryconsumptionpath,
+            '--app-exe', traits.apptorun,
+            '--parse-only',
```

Comment:
Should this be controlled by a test trait? In the future if we end up running memory tests that run locally we would not want to default to `--parse-only`